src/compiler/utilities.ts

```diff
@@ -3298,6 +3297,10 @@ namespace ts {
         return false;
     }
 
+    export function isJSDocTypeReference(node: TypeReferenceType): node is TypeReferenceNode {

```

Comment:
isInJavaScriptFile may not be sufficient because JSDoc types can appear anywhere in TS or JS files, inside or outside JSDoc. For example:

```js
/** @type {...number} */
var x;
var y: ...number; // annotations are errors in JS file, but still fully checked and treated as number[]
```

and 

```ts
/** @type {...number} */ // actually, this isn't checked.
var x;
var y: ...number; // ... is an error in TS file, but still fully checked and treated as number[]
```